frontend/src/api/calls/skillTypeCalls.ts

```diff
@@ -42,3 +43,25 @@ export async function getSkillTypeFromSkill(skill: IUserSkill | IProjectSkill):
 export function getAllSkillTypesFromLinks(url: string): Promise<ISkillType[]> {
     return <Promise<ISkillType[]>>getAllEntitiesFromLinksUrl(url, skillTypeCollectionName);
 }
+
+export async function getSkillTypeWithName(skillName: string): Promise<ISkillType> {
+    let skills = await getAllSkillTypesFromLinks(apiPaths.skillTypes);
+    const skill = skills.find((skill) => skill.name === skillName.slice(0, 20));
+    let skillType: ISkillType;
+    if (skill == undefined) {
+        skillType = await createNewSkill(skillName);
+    } else {
+        skillType = skill;
+    }
+    return skillType;
+}
+
+export function getRandomColor() {
+    let color = Math.floor(Math.random() * 16777216).toString(16);
+    return "#000000".slice(0, -color.length) + color;
+}
+
+export async function createNewSkill(skillName: string): Promise<ISkillType> {
+    const skill = new SkillType(skillName.slice(0, 20), getRandomColor());
```

Comment:
This check shouldn't be needed? We can verify this length with the user input form? -> If the user hacks their way through, the back end will handle this.